### PR TITLE
Integrate repo with SonarQube

### PR DESCRIPTION
**Pull request by chudilka1**

### CODE REVIEW

File: .github/workflows/lint.yaml [hunk 1]
```diff
@@ -26,9 +27,24 @@ jobs:
...
           go-version: ${{ needs.tools.outputs.go-version }}
       - name: golangci-lint ${{ needs.tools.outputs.golangci-lint-version }}
-        uses: golangci/golangci-lint-action@v3.4.0
+        ##
+        # XXX: change this to the official action once multiple --out-format args are supported.
+        # See: https://github.com/golangci/golangci-lint-action/issues/612
+        ##
+        uses: smartcontractkit/golangci-lint-action@54ab6c5f11d66a92d14c3f7cc41ea13f676644bd # feature/multiple-output-formats-backup
+        with:
+          version: v1.52.1
```
Comment (tateexon) [inline .github/workflows/lint.yaml:36]:
```suggestion
          version: v${{ needs.tools.outputs.golangci-lint-version }}
```

Comment (tateexon) [inline .github/workflows/lint.yaml:36]:
```suggestion
          version: v${{ needs.tools.outputs.golangci-lint-version }}
```

File: .github/workflows/lint.yaml [hunk 2]
```diff
@@ -26,9 +27,24 @@ jobs:
         with:
           go-version: ${{ needs.tools.outputs.go-version }}
       - name: golangci-lint ${{ needs.tools.outputs.golangci-lint-version }}
-        uses: golangci/golangci-lint-action@v3.4.0
+        ##
+        # XXX: change this to the official action once multiple --out-format args are supported.
+        # See: https://github.com/golangci/golangci-lint-action/issues/612
+        ##
+        uses: smartcontractkit/golangci-lint-action@54ab6c5f11d66a92d14c3f7cc41ea13f676644bd # feature/multiple-output-formats-backup
```
Comment (tateexon) [inline .github/workflows/lint.yaml:34]: Why did we make our own outdated fork of golangci-lint?

**Review by tateexon — CHANGES_REQUESTED** [review]

File: .github/workflows/lint.yaml (hunk 1 above)
Comment (chudilka1) [reply .github/workflows/lint.yaml:36]: There is a comment about why namely this version is being used, there is no way to provide the needed output format...

**Review by tateexon — APPROVED** [review]

**Review by skudasov — APPROVED** [review]

**Pull request merged**